case-lib/opt.sh

```diff
@@ -115,27 +117,11 @@ func_opt_parse_option()
         fi
     done
     
-    [ ${OPT_VALUE_lst['h']} -eq 1 ] && _func_opt_dump_help
-    # verify & check all option values
-    [[ $(declare -p |grep "[[:space:]]OPT_VALUE_lst=\"\"") ]] && echo "Missing parameter/script default value" && _func_opt_dump_help
-    # record the full parameter to the cmd
-    if [ ! -f $LOG_ROOT/cmd.txt ]; then
-        local cmd="$0" opt
-        for opt in ${!OPT_OPT_lst[@]}
-        do
-            [[ $opt == "h" ]] && continue
-            if [ ${OPT_PARM_lst[$opt]} -eq 1 ]; then
-                cmd="$cmd -$opt ${OPT_VALUE_lst[$opt]}"
-            else
-                if [ $(echo '!'${OPT_VALUE_lst[$opt]}|bc) -eq 1 ]; then
-                    cmd="$cmd -$opt"
-                fi
-            fi
-        done
-        echo $cmd >> $LOG_ROOT/cmd.txt
-        local git_path=$(dirname $0)
-        echo "Branch: "$(git -C $git_path branch |grep '^*') >> $LOG_ROOT/version.txt
-        echo "Commit: "$(git -C $git_path log --oneline -1) >> $LOG_ROOT/version.txt
+    [ "${OPT_VALUE_lst['h']}" -eq 1 ] && _func_opt_dump_help
+    # record the git version
+    if [[ ! -f "$LOG_ROOT/version.txt" ]] && [[ -f "$SCRIPT_HOME/.git/config" ]]; then
+        echo -e "Branch:\n$(git -C "$SCRIPT_HOME" branch |sed 's/^/\t/g')" >> "$LOG_ROOT/version.txt"
+        echo -e "Commit:\n$(git -C "$SCRIPT_HOME" log --branches --oneline -n 5)" >> "$LOG_ROOT/version.txt"
```

Comment:
See SC2005 and SC2039
```shell
{
  printf 'Branch:'
  git -C "$SCRIPT_HOME" branch |sed 's/^/\t/g'
  printf 'Commit:'
  git -C "$SCRIPT_HOME" log --branches --oneline -n 5

} >> "$LOG_ROOT/version.txt"
```

Reply:
sorry, in my environment: shellcheck v0.7.0 without this problem

Reply:
Sorry that was unclear and inaccurate, I meant have a look at:
https://github.com/koalaman/shellcheck/wiki/SC2005
https://unix.stackexchange.com/questions/65803/why-is-printf-better-than-echo

Reply:
got it, how about this style?
```
cat << END >"$LOG_ROOT/version.txt"
output some thing
$(some command)
END
```

Reply:
This drops the exit status of `some command`.

If you really really want to use `cat <<EOF` for some reason then it should be something like this:
```shell
local val
val=$(some command) # SC2155
cat <<END >"$LOG_ROOT/version.txt"
output something
${val}
END
```
... but that's a lot code more code than simply running `some command`